src/Tasks/CreateManifestResourceName.cs

```diff
@@ -161,8 +163,19 @@ CreateFileStream createFileStream
                     // If opted into convention and no DependentUpon metadata and is a resx file, reference "<filename>.cs" if it exists.
                     if (isResxFile && UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))
                     {
+                        // Assume that by convention the expected file name is "<filename>.cs"
                         string conventionDependentUpon = Path.ChangeExtension(Path.GetFileName(fileName), SourceFileExtension);
 
+                        // Verify that the file name didn't have a culture associated with it. Ex: "<filename>.<culture>.resx" If we don't strip the culture we look for TestComponent.de.cs, which we don't want.
+                        if (resourceFile.GetMetadata("WithCulture") == "true")
+                        {
+                            string culture = resourceFile.GetMetadata("Culture");
+                            int indexJustBeforeCulture = fileNameWithoutExtension.Length - culture.Length - 1;
+
+                            //strip the culture from the name, append the ".cs" extension, now we have "<filename>.cs", this is the file resourceFile is dependent upon
+                            conventionDependentUpon = fileNameWithoutExtension.Substring(0, indexJustBeforeCulture) + ".cs";
```

Comment:
```suggestion
                            conventionDependentUpon = fileNameWithoutExtension.Substring(0, indexJustBeforeCulture) + SourceFileExtension;
```

right? Can you double-check with a VB project?